manage_people/migrations/0002_manage_people_uses_coursemanager_user_roles.py

```diff
@@ -1,137 +0,0 @@
-    ManagePeopleRole = apps.get_model('manage_people', 'ManagePeopleRole')
-    fields = ('user_role_id', 'canvas_role_label', 'xid_allowed')
-    with transaction.atomic():  # wrap all the inserts in a transaction
-        for values in MANAGE_PEOPLE_ROLE_DATA:
-            ManagePeopleRole.objects.create(**dict(zip(fields, values)))
-
-
-def update_school_allowed_role(apps, schema_editor):
-    """
-    For each role in school_allowed_role, we need to:
-    * get the canvas_roles entry it refers to
-    * read the canvas_roles.canvas_role value
-    * find the matching user_role_id for that canvas_role from our canned
-      manage_people_role data we're populating during this migration
-    * replace the existing canvas_role_id with the new user_role_id
-
-    NOTE: this expects to be run before the schema change to rename
-          school_allowed_role.canvas_role_id to
-          school_allowed_role.user_role_id
-    """
-    SchoolAllowedRole = apps.get_model('manage_people', 'SchoolAllowedRole')
-    with transaction.atomic():  # wrap all the updates in a transaction
-        for sar in SchoolAllowedRole.objects.all():
-            sar.canvas_role_id = SCHOOL_ALLOWED_ROLE_ID_UPDATE_MAP[sar.canvas_role_id]
-            sar.save()
-
-
-def revert_school_allowed_role(apps, schema_editor):
-    """
-    Reverts the changes made in update_school_allowed_role.
-
-    NOTE: this expects to be run after the schema change to rename
-          school_allowed_role.user_role_id to
-          school_allowed_role.canvas_role_id
-    """
-    SchoolAllowedRole = apps.get_model('manage_people', 'SchoolAllowedRole')
-    with transaction.atomic():  # wrap all the updates in a transaction
-        for sar in SchoolAllowedRole.objects.all():
-            sar.canvas_role_id = SCHOOL_ALLOWED_ROLE_ID_REVERT_MAP[sar.canvas_role_id]
-            sar.save()
-
-
-class Migration(migrations.Migration):
-    dependencies = [
-        ('manage_people', '0001_initial'),
-    ]
-
-    operations = [
-        migrations.CreateModel(
-            name='ManagePeopleRole',
-            fields=[
-                ('user_role_id', models.IntegerField(primary_key=True)),
-                ('canvas_role_label', models.CharField(unique=True, max_length=30)),
-                ('xid_allowed', models.BooleanField(default=False)),
-            ],
-            options={
-                'db_table': 'manage_people_role',
-            },
-        ),
-        migrations.RunPython(
-            code=populate_manage_people_role,
-            reverse_code=migrations.RunPython.noop,
-        ),
-        migrations.RunPython(
-            code=update_school_allowed_role,
-            reverse_code=revert_school_allowed_role,
-        ),
-        migrations.RenameField(
-            model_name='schoolallowedrole',
-            old_name='canvas_role_id',
-            new_name='user_role_id',
-        ),
-        migrations.AlterUniqueTogether(
```

Comment:
In the current state Django won't respect the unique together that the SchoolAllowedRole model specifies.

Running `python manage.py migrate` gives:

```
  Your models have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
```

And:

```
  (canvas_course_admin_tools)vagrant@vagrant-ubuntu-trusty-64:~/tlt/canvas_course_admin_tools(task/sapnamysore/tlt-1870/cleanup-migrations) $ python manage.py makemigrations
Migrations for 'manage_people':
  0003_auto_20160518_1418.py:
    - Alter field user_role on schoolallowedrole
    - Alter unique_together for schoolallowedrole (1 constraint(s))
```

For reference, here's what Django creates.

```

class Migration(migrations.Migration):

    dependencies = [
        ('manage_people', '0002_mp_school_allowed_roles'),
    ]

    operations = [
        migrations.AlterField(
            model_name='schoolallowedrole',
            name='user_role',
            field=models.ForeignKey(to='manage_people.ManagePeopleRole'),
        ),
        migrations.AlterUniqueTogether(
            name='schoolallowedrole',
            unique_together=set([('school_id', 'user_role')]),
        ),
    ]

```

After running that migration, Django observes the unique together constraint.